[llvm-lit][test][NFC] Moved cat command tests into separate lit test file

connieyzhu opened the pull request:
This patch separates the lit tests that check for the functionality of lit's built-in cat command into its own test file and folder. This is a prerequisite for https://github.com/llvm/llvm-project/pull/101530.

connieyzhu committed:
[llvm-lit][test][NFC] Moved cat command tests into separate lit test
file

This patch separates the lit tests that check for the functionality of
lit's built-in cat command into its own test file and folder.

connieyzhu commented:
CC: @ilovepi @petrhosek @jh7370

ilovepi commented on llvm/utils/lit/tests/Inputs/shtest-cat/cat.txt:
```diff
@@ -0,0 +1,83 @@
+# Check cat command with single file.
+#
+# RUN: rm -rf %T/testCat
```
I know these are the existing test lines, but we’ve deprecated %T. Instead use %t. I think you also probably do need the testCat subdirectory either.

jh7370 replied:
I'd normally agree with you, but as the intent is to make various modifications to improve these existing tests in a follow-up, I think it makes sense to defer this change to that patch too, to make it clearer what actually changed.

I'm ambivalent about the extra directory. Happy either way on that one.

connieyzhu replied:
I can make these changes in my follow-up patch. For changing %T to %t, how would I get the filepath of %t? I tried looking through the lit documentation and existing tests and I couldn't find a clear answer. Is it something like %t/dir?

connieyzhu replied:
Yes that does - so %t is just a more versatile option that encapsulates %T, which is why %T is not used anymore?

jh7370 replied:
As I understand it, the main problem with `%T` is that it isn't unique to the individual test, so you could end up with two tests trying to create `%T/foo.o` and you'd end up with race conditions if the two were run in parallel. The same goes for directories. It may be possible to avoid these name clashes, but in practice, doing so takes more mental power than is really necessary.

ilovepi commented on llvm/utils/lit/tests/Inputs/shtest-cat/check_path.py:
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env python
```
Instead of adding this file, can we move the existing copy up a directory?that is probably it’s own patch.

connieyzhu committed:
[llvm-lit][test][NFC] Moved check_path.py to be used by multiple
directories

This patch moves check_path.py into the main Inputs directory so that it
can be used by both shtest-cat and shtest-shell.

jh7370 commented on llvm/utils/lit/tests/shtest-cat.py:
```diff
@@ -0,0 +1,20 @@
+# CHECK: # | [Errno 2] No such file or directory: 'temp1.txt'
+# CHECK: # error: command failed with exit status: 1
+
+# CHECK: PASS: shtest-cat :: cat.txt ({{[^)]*}})
```
Nit: add a new line at EOF while you're creating it.

jh7370 commented on llvm/utils/lit/tests/shtest-cat.py:
```diff
@@ -0,0 +1,20 @@
+## Check the cat command.
```
Should this test check the passed/failed summary like the shtest-shell test does?

ilovepi replied:
Probably. It's likely to catch out tests that are added, but didn't update this file correctly. I'd expect it would be possible to miss them otherwise.

jh7370 reviewed (commented):
What do you think about removing `shtest` from the test directory names? I'm not convinced about its value to the test description.

connieyzhu committed:
[llvm-lit][test][NFC] Added pass/fail summary to cat testing

connieyzhu commented:
> What do you think about removing `shtest` from the test directory names? I'm not convinced about its value to the test description.

@jh7370 From what I can tell, the test directories with `shtest` are involved with some sort of shell command in their tests, while those without it don't use shell commands (can't say this is 100% true, but just from scanning several files). I do think that having the description is pretty helpful in knowing which tests will involve lit's built-in commands, which I will need to reference as I write tests for other features I'm planning on implementing. So I don't think it's worth removing from every single test directory and test file (there's >15 of them).

ilovepi commented on llvm/utils/lit/tests/Inputs/shtest-cat/cat.txt:
```diff
@@ -0,0 +1,83 @@
+# Check cat command with single file.
```
```suggestion
## Test cat command with a single file.
```

connieyzhu replied:
I can make this change in my follow-up patch when I update all the content.

jh7370 commented on llvm/utils/lit/tests/shtest-cat.py:
```diff
@@ -17,4 +17,8 @@
\ No newline at end of file
+# CHECK: PASS: shtest-cat :: cat.txt ({{[^)]*}})
+
+# CHECK: Total Discovered Tests: 3
+# CHECK: Passed: 1 {{\([0-9]*\.[0-9]*%\)}}
+# CHECK: Failed: 2 {{\([0-9]*\.[0-9]*%\)}}
```
Can a combination of `CHECK-NEXT` and/or `CHECK-EMPTY` be used in this test file? This helps with diagnosing issues, if the match fails.

jh7370 reviewed (approved):
One suggestion. LGTM with or without it.

connieyzhu committed:
[llvm-lit] Made syntax changes

connieyzhu committed:
Modified CHECK directive syntax
